gsy_framework/data_classes.py

```diff
@@ -173,17 +218,21 @@ def from_dict(offer: Dict) -> "Offer":
             energy=offer["energy"],
             price=offer["energy"] * offer["energy_rate"],
             original_price=offer.get("original_price"),
-            seller=offer.get("seller"),
-            seller_origin=offer.get("seller_origin"),
-            seller_origin_id=offer.get("seller_origin_id"),
-            seller_id=offer.get("seller_id"),
+            seller=TraderDetails(
+                name=offer["seller"]["name"],
+                origin=offer["seller"]["origin"],
+                origin_uuid=offer["seller"]["origin_uuid"],
+                uuid=offer["seller"]["uuid"],
+            ),
             attributes=offer.get("attributes"),
             requirements=offer.get("requirements"))
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: "Offer") -> bool:
         return (self.id == other.id and
+                isclose(self.energy_rate, other.energy_rate, rel_tol=FLOATING_POINT_TOLERANCE) and
+                isclose(self.energy, other.energy, rel_tol=FLOATING_POINT_TOLERANCE) and
+                isclose(self.price, other.price, rel_tol=FLOATING_POINT_TOLERANCE) and
```

Comment:
How come we did not check for these before or rather: Why is it needed now?

Reply:
Simply because we have never stumbled upon this error till now. Admittedly, I observed the behavior only for the energy_rate (because it executes the division price / energy which can lead potentially to a floating point arithmetic error), however I thought to add also the energy and the price in the comparison in order to enforce the same floating point precision to all the numerical attributes.